x-pack/plugins/security_solution/cypress/tasks/create_new_rule.ts

```diff
@@ -670,17 +672,22 @@ export const selectNewTermsRuleType = () => {
 export const waitForAlertsToPopulate = async (alertCountThreshold = 1) => {
   cy.waitUntil(
     () => {
+      cy.log('Waiting for alerts to appear');
       refreshPage();
-      return cy
-        .get(SERVER_SIDE_EVENT_COUNT)
-        .invoke('text')
-        .then((countText) => {
-          const alertCount = parseInt(countText, 10) || 0;
-          return alertCount >= alertCountThreshold;
-        });
+      return cy.root().then(($el) => {
+        const emptyTableState = $el.find('[data-test-subj="alertsStateTableEmptyState"]');
+        if (emptyTableState.length > 0) {
+          cy.log('Table is empty', emptyTableState.length);
+          return false;
+        }
+        const countEl = $el.find(ALERTS_TABLE_COUNT);
+        const alertCount = parseInt(countEl.text(), 10) || 0;
+        return alertCount >= alertCountThreshold;
+      });
     },
     { interval: 500, timeout: 12000 }
   );
+  waitForAlerts();
```

Comment:
Is this necessary? I believe we could delete this line since we have already waited for the alerts to be present.

Reply:
@MadameSheema , Actually it is needed because in first step we are just making sure that alerts appear. But it is not sure if they are ready for any action to be performed because there are various loading indicators. 

`waitForAlerts` makes sure that all of those loading indicators are also gone and then the table is ready for any action to be performed.